build.gradle

```diff
@@ -824,8 +824,19 @@ project(':iceberg-pig') {
 }
 
 project(':iceberg-nessie') {
-  test {
-    useJUnitPlatform()
+  if (JavaVersion.current().isJava11Compatible()) {
+    test {
+      useJUnitPlatform()
+    }
+  } else {
+    // Do not test Nessie against Java 8, because in-JVM testing requires Nessie server components,
+    // which require Java 11+.
+    test {
+      enabled = false
+    }
+    compileTestJava {
+      enabled = false
```

Comment:
Maybe still compile tests to detect accidental syntax errors?

Reply:
Still compiles w/ Java 11+. Once we (Nessie) also have bumped the server-side code to Java 11, it would start to fail.